custom_components/adaptive_lighting/switch.py

```diff
@@ -238,57 +237,136 @@ def _split_service_data(service_data, adapt_brightness, adapt_color):
-        if data[CONF_TURN_ON_LIGHTS] or is_on(hass, light):
-            await switch._adapt_light(  # pylint: disable=protected-access
-                light,
-                data[CONF_TRANSITION],
-                data[ATTR_ADAPT_BRIGHTNESS],
-                data[ATTR_ADAPT_COLOR],
-                data[CONF_PREFER_RGB_COLOR],
-                force=True,
-                context=switch.create_context("service", parent=service_call.context),
+    data = service_call.data
+    lights = data[CONF_LIGHTS]
+    switch_unique_id = data.get("entity_id")
+    switch = None
+    if not lights and not switch_unique_id:
+        # alternatively just loop and call the service for every switch.
+        _LOGGER.debug(
+            "adaptive-lighting: If you were trying to adapt every single light on every single"
+            " switch, please inform the developers at"
+            " https://github.com/basnijholt/adaptive-lighting of your use case."
+            " Currently you must pass either the adapt main switch, or the lights to every"
+            " service call. adaptive-lighting will now raise a soft error for you in the meantime."
+        )
+        _LOGGER.error(
+            "bad service data to adaptive-lighting service call - "
+            "you must pass either an entity_id or a light. Service data:\n%s",
+            service_call.data,
+        )
+        raise ValueError(
+            "adaptive-lighting: No switch or light was passed to service call."
+        )
+
+    if switch_unique_id is not None:
+        if len(switch_unique_id) > 1 and lights:
+            _LOGGER.error(
+                "bad service data: cannot pass multiple switch entities while also passing"
+                "any lights. Got %s",
+                service_call.data,
+            )
+            raise ValueError(
+                "adaptive-lighting: Multiple switches passed with lights argument"
             )
+        switch = []
+        for unique_id in switch_unique_id:
+            ent_reg = entity_registry.async_get(hass)
+            ent_entry = ent_reg.async_get(unique_id)
+            # `AttributeError: 'NoneType' object has no attribute 'config_entry_id'` fix later maybe
+            config_id = ent_entry.config_entry_id
+            _LOGGER.debug(
+                "_parse_service_args: switch: %s, ent_reg: %s, ent_entry: %s, config_id: %s",
+                switch,
+                ent_reg,
+                ent_entry,
+                config_id,
+            )
+            switch.append(hass.data[DOMAIN][config_id]["instance"])
+    elif lights:
+        # all_adapt_switches = integration_entities(hass, DOMAIN)
+        # No need to check if light is found in multiple switches, that's a user error.
+        config_entries = hass.config_entries.async_entries(DOMAIN)
+        found = False
+        for config in config_entries:
+            if found:
+                break
+            _LOGGER.debug("entry_id: %s", config.entry_id)
```

Comment:
Some of these `_LOGGER.debug` statements seem a little uninformative for debugging in the future. I would say, either remove them, or make them a bit more informative.

Reply:
Looks like you've already done so.